Controller/Hosted/RetryPayment.php

```diff
@@ -67,7 +67,7 @@ protected function getRedirectResponse($order)
             $customData['customer_id'] = $order->getCustomerId();
         }
 
-        $requestData = $isMultishipping ? $this->processMultishipping($order) : $this->hostedContext->hostedCheckoutBuilder->getData($order->getIncrementId(), $amount, $customData);
+        $requestData = $isMultishipping ? $this->processMultishipping($order) : $this->hostedContext->hostedCheckoutBuilder->getData((int) $order->getQuoteId(), $order->getIncrementId(), $amount, $customData);
```

Comment:
What necessitates the cast here? If it's guaranteed to be an integer, why isn't it returned like that from `getQuoteId()`?

- [ ] Confirm quote id casting

Reply:
For some reason i had an error where it would return the integer as a string (eg: "32" instead of 32) so I added a explicit cast to avoid this. Magento 2 has "magic" getter and setters on it's base models, unless they explicitly define the getters and setters it will be of any type pretty much.